### PR TITLE
Ft strictdi

### PR DESCRIPTION
**Pull request by forrestbthomas**
Fixes the tests and adds back strict-di

### CODE REVIEW

File: generators/app/templates/_karma.conf.js [hunk 1]
```diff
@@ -18,7 +18,8 @@ function listFiles() {
       path.join(conf.paths.src, '/app/**/*.js'),
       path.join(conf.paths.src, '/**/*.spec.js'),
       path.join(conf.paths.src, '/**/*.mock.js'),
-      path.join(conf.paths.src, '/**/*.html')
+      path.join(conf.paths.src, '/**/*.html'),
+      path.join(conf.paths.bower, '/**/*.min.js')
```
Comment (michikono) [inline generators/app/templates/_karma.conf.js]: this is tricky -- not all bower distributions have `.min` files. Bower specifically has a configuration entry to describe this: https://github.com/bower/bower.json-spec (see "main")

File: generators/app/templates/_karma.conf.js [hunk 2]
```diff
@@ -27,6 +28,8 @@ module.exports = function(config) {
   var configuration = {
     files: listFiles(),
 
+    exclude: [path.join(conf.paths.bower, '/font-awesome/**/*.min.js')],
```
Comment (michikono) [inline generators/app/templates/_karma.conf.js]: see previous note

File: generators/app/templates/client/app/_app.core.module.js [hunk 3]
```diff
@@ -31,6 +31,8 @@
     $locationProvider.html5Mode(true).hashPrefix('!');
   }
 
+  configure.$inject = ['$locationProvider'];
```
Comment (michikono) [inline generators/app/templates/client/app/_app.core.module.js]: You don't need to provide this so long as we're using a good ng-annotate script along with the `@ngInject` annotation. I've purposely avoided this to keep the code cleaner. See this for more details: https://github.com/olov/ng-annotate#explicit-annotations-with-nginject

File: generators/app/templates/gulp/unit-tests.js [hunk 4]
```diff
@@ -11,7 +11,9 @@ function runTests (singleRun, done) {
     configFile: path.join(__dirname, '/../karma.conf.js'),
     singleRun: singleRun,
     autoWatch: !singleRun
-  }, done);
+  }, function() {
```
Comment (michikono) [inline generators/app/templates/gulp/unit-tests.js:14]: I believe this is functionally identical (but more verbose) to the original code?

Comment (forrestbthomas) [inline generators/app/templates/gulp/unit-tests.js:14]: Here is the SO post re: this issue

http://stackoverflow.com/questions/26614738/issue-running-karma-task-from-gulp

**Pull request merged**